core/sup_tree_core/executor_pool/memcache_writer.ex

```diff
@@ -0,0 +1,100 @@
+# Copyright(c) 2015-2018 ACCESS CO., LTD. All rights reserved.
+
+use Croma
+
+defmodule AntikytheraCore.ExecutorPool.MemcacheWriter do
+  @moduledoc """
+  A `GenServer` to manage memcache for each executor pool.
+  """
+
+  @timeout         300_000
+  @max_records_num 10
+
+  use GenServer
+  alias AntikytheraCore.Ets.Memcache
+  alias AntikytheraCore.ExecutorPool.RegisteredName, as: RegName
+  alias Antikythera.ExecutorPool.Id, as: EPoolId
+
+  def start_link(name, epool_id) do
+    GenServer.start_link(__MODULE__, epool_id, [name: name])
+  end
+
+  @impl true
+  def init(epool_id) do
+    {:ok, %{epool_id: epool_id, records: %{}, expire_at_set: :gb_sets.new()}, @timeout}
+  end
+
+  @impl true
+  def handle_call({:write, key, value, lifetime, ratio}, _from, state) do
+    new_state =
+      state
+      |> evict_expired_records(System.monotonic_time(:milliseconds))
+      |> do_write(key, value, lifetime, ratio)
+      |> evict_head_records_if_exceeds_limit()
+    {:reply, :ok, new_state, @timeout}
+  end
+
+  @impl true
+  def handle_info(:timeout, state) do
+    new_state = evict_expired_records(state, System.monotonic_time(:milliseconds))
+    {:noreply, new_state, @timeout}
+  end
+
+  defp evict_expired_records(%{records: records, expire_at_set: expire_at_set, epool_id: epool_id} = state, current_time) do
+    if :gb_sets.is_empty(expire_at_set) do
+      state
+    else
+      {{expire_at, key}, new_expire_at_set} = :gb_sets.take_smallest(expire_at_set)
```

Comment:
Oh, my bad.
Very sorry for writing a comment in the ticket about `take_largest`.

Reply:
No problem. I should notice at first of writing.